src/java.base/share/classes/jdk/internal/foreign/MemorySessionImpl.java

```diff
@@ -73,9 +77,27 @@
         }
```

Comment:
The above `catch` clause should only catch `Exception`s, not `Throwable`s, as the latter would hide VM errors such as `StackOverflowError` or `OutOfMemoryError`.